### ISSUE TITLE
Support domain socket

### ISSUE THREAD

**Issue by daixiang0** [dapr/java-sdk#600]
## Describe the proposal

Ref https://github.com/dapr/dapr/issues/2864

## Release Note

RELEASE NOTE: **ADD** support domain socket.

**Comment by mukundansundar** [comment]
This needs a change in dapr/cli before this can be picked up.

**Comment by mthmulders** [comment]
I believe domain socket support was added to Java in version 16 with [JEP-380](https://openjdk.java.net/jeps/380). As a consequence, if we want to support older versions of Java as well, we should have a fallback strategy to HTTP or gRPC.

**Comment by YEthYuan** [comment]
Hi, my name is Ye Yuan and will attend Open Source Day '23. I'm interested in this issue!

**Comment by YEthYuan** [comment]
> I believe domain socket support was added to Java in version 16 with [JEP-380](https://openjdk.java.net/jeps/380). As a consequence, if we want to support older versions of Java as well, we should have a fallback strategy to HTTP or gRPC.

I've reviewed the implementation for go sdk (https://github.com/dapr/go-sdk/pull/206/files). So since we'd like to support older Java versions which don't support UDS, for Java sdk, we should try HTTP or gRPC rather than directly follow the implementation for go sdk, right?

**Comment by artursouza** [comment]
I don't think we need to match the Go SDK implementation. IMO, the connection to the sidecar should happen transparently to the user based on the values of DAPR_HTTP_ENDPOINT and DAPR_GRPC_ENDPOINT env variables.

**Comment by YEthYuan** [comment]
thanks! good to know
